Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue2577.cs

```diff
@@ -49,5 +49,43 @@ public DetailPage()
 				Content = button;
 			}
 		}
+
+#if UITEST
+		[Test]
+		public void Issue2577Test()
+		{
+			RunningApp.WaitForElement("NavigateButton");
+			RunningApp.Tap("NavigateButton");
+
+			RunningApp.WaitForElement("ToggleBackButton");
+
+			RunningApp.Screenshot("Hamburger menu icon is visible");
+
+			AppResult[] items = RunningApp.Query("OK");
+			Assert.AreNotEqual(items.Length, 0);
+			RunningApp.Tap("OK");
+			RunningApp.WaitForElement("MasterList");
+
+			RunningApp.Screenshot("Flyout menu is showing");
+			
+			RunningApp.SwipeRightToLeft();
+			RunningApp.WaitForNoElement("MasterList");
+
+			RunningApp.Tap("ToggleBackButton");
+
+			items = RunningApp.Query("OK");
+			Assert.AreEqual(items.Length, 0);
+
+			RunningApp.Screenshot("Back arrow is showing");
+
+			var backArrow = RunningApp.Query(e => e.Class("AppCompatImageButton")).First(e => e.Label == null);
+
+			RunningApp.TapCoordinates(backArrow.Rect.CenterX, backArrow.Rect.CenterY);
```

Comment:
Is this necessary? or you can use simple `RunningApp.Back()` ?

Reply:
I think `Back` uses the hardware back button, and going back through the actual arrow here is key to this test